R/object_usage_linter.R

```diff
@@ -100,8 +100,8 @@ object_usage_linter <- function(interpret_glue = TRUE, skip_with = TRUE) {
       lintable_symbols <- xml2::xml_find_all(
         fun_assignment,
         "
-        descendant::SYMBOL
-        | descendant::SYMBOL_FUNCTION_CALL
+        descendant::SYMBOL[not(ancestor::expr[OP-TILDE])]
+        | descendant::SYMBOL_FUNCTION_CALL[not(ancestor::expr[OP-TILDE])]
         | descendant::SPECIAL
```

Comment:
Could this also appear in a formula?

Reply:
I think any infix operator used in a formula has to be defined elsewhere, so no. trying to thing of an example...

```r
foo <- function() {
  lm(y ~ X %**% Z, data = X %**% Y)
}
```

If `checkUsage` is dinging `%**%`, it should also be dinged in the formula.

The only thing I can think of is where the operator has some special definition inside the formula, which doesn't happen in base since only `%in%` has special interpretation:

https://stat.ethz.ch/R-manual/R-devel/library/stats/html/formula.html

I guess it's conceivable for other downstreams to define special interpretation methods for formulas, but because `checkUsage()` ignores formulas entirely (IIUC), it would have to be that _both_ (1) `%**%` has a special interpretation _and_ (2) it's being used incorrectly outside of a formula.

That seems pretty out there...